### PR TITLE
Added separate user projection in query help for rest api

### LINKED ISSUE #2764: REST API should not be based on Hybrid properties
Currently REST API is based on hybrid properties like the user_email defined in the DbNode table in SQLA.

This allows us to create queries where we may project node properties and also the user_email and implicitly to do the join of the DbNode with the DbUser to provide the user email.

This is a nice feature but it is a feature that it is defined at the SQLA model (not in the Django model) and we should not make assumptions that such feature exist at the REST API which is a few levels above the backends.

In general the REST API queries should change from 

```
qh = {'project': {
    'calculations': ['id', 'label', 'node_type', 'ctime', 'mtime', 'uuid', 'user_id', 'user_email',
                     'attributes', 'extras']},
      'path': [{'tag': 'calculations', 'entity_type': 'process.calculation.calcjob.CalcJobNode.'}],
      'order_by': {'calculations': {}}, 'filters': {}}
```

to

```
qh = {
    'project':
        {
            'calc': ['id', 'label', 'node_type', 'ctime', 'mtime', 'uuid', 'user_id', 'attributes', 'extras'],
            'user_1': ['email']
        },
    'path':
        [
            {'entity_type': 'process.calculation.calcjob.CalcJobNode.', 'tag': 'calc'},
            {'edge_tag': 'calc--user_1', 'joining_keyword': 'with_node', 'entity_type': 'user',
             'joining_value': 'calc', 'tag': 'user_1'}
    ],
    'order_by': {'calc': {}}
```

This issue became visible after the removal of the dummy_model (for the querying of Django backrend). Work which is done to support JSONB in Djnago

### PR DESCRIPTION
**Pull request by waychal**
Resolves #2764

### CODE REVIEW

File: aiida/restapi/translator/node.py [hunk 1]
```diff
@@ -268,7 +269,7 @@ def _get_content(self):
             return {}
 
         # otherwise ...
-        node = self.qbobj.first()[0]
+        node = self.qbobj.all()[0][1]
```
Comment (szoupanos) [inline aiida/restapi/translator/node.py]: We should see with Leonid why this is not possible anymore.
If we don't find a solution, a ticket should be opened.

**Comment by szoupanos** [comment]
This should be merged after it is tested with the django_jsonb branch where the hybrid properties are removed.

I will need help from @sphuber to rebase the django_jsonb branch since after his last PRs I have conflicts that are not obvious to resolve.

Then we can test this PR with the rebased django_jsonb and merge it to develop.

File: aiida/restapi/translator/node.py (hunk 1 above)
Comment (sphuber) [reply aiida/restapi/translator/node.py]: The issue describing this #2808 I have fixed in PR #2824 so this change can now be reverted

**Comment by sphuber** [comment]
I have tested this with the `django_jsonb` branch and confirmed that it fixes the tests. I will merge this in `develop` and then rebase `django_jsonb` which will allow the REST API tests to be re-enabled.

**Pull request merged**